Add helm chart for cloudwatch

bnrjee opened the pull request:
*Issue #, if available:*

*Description of changes:*
Modified the cloudwatch-agent part of this [file](https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/latest/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/quickstart/cwagent-fluentd-quickstart.yaml) and turned into a helm chart. Also made changes in service account creation logic, daemonset permissions etc.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

<!-- If this is a security issue, please do not discuss on GitHub. Please report any suspected or confirmed security issues to AWS Security https://aws.amazon.com/security/vulnerability-reporting/ -->

vivek-koppuru commented:
Is this a helm chart we need to construct on our own or do something like what we are planning to do with prometheus where we check out the repository and then build and upload the helm charts?

bnrjee commented:
> Is this a helm chart we need to construct on our own or do something like what we are planning to do with prometheus where we check out the repository and then build and upload the helm charts?

We modified the helm chart.

vivek-koppuru commented:
> > Is this a helm chart we need to construct on our own or do something like what we are planning to do with prometheus where we check out the repository and then build and upload the helm charts?
> 
> We modified the helm chart.

Can you list in the description what the upstream is and how we modified it then? Just want to be careful about maintain more and more charts that we can get from upstream, and potentially start thinking about using stuff like Kustomize for our modifications of a helm chart.

bnrjee commented:
> > > Is this a helm chart we need to construct on our own or do something like what we are planning to do with prometheus where we check out the repository and then build and upload the helm charts?
> > 
> > 
> > We modified the helm chart.
> 
> Can you list in the description what the upstream is and how we modified it then? Just want to be careful about maintain more and more charts that we can get from upstream, and potentially start thinking about using stuff like Kustomize for our modifications of a helm chart.

I think, I should rephrase my answer. It wasn't a helm chart upstream.
https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/latest/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/quickstart/cwagent-fluentd-quickstart.yaml

I took out the cloud-watch agent part of this yaml and created a helm chart out of it with some modifications. There was a [PR](https://github.com/aws/eks-distro-build-tooling/pull/34) created for this before. I lost that remote branch, so created this one.

vivek-koppuru commented on helm-charts/stable/cloudwatch-agent/templates/_helpers.tpl:
```diff
@@ -0,0 +1,38 @@
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
```
Let's add an app selector like we do here as it is useful for when we add integration testing: https://github.com/aws/eks-distro-build-tooling/blob/main/helm-charts/stable/amazon-eks-pod-identity-webhook/templates/_helpers.tpl#L49

vivek-koppuru replied:
Then that way, the labels can be added to all the templates as such: https://github.com/aws/eks-distro-build-tooling/blob/main/helm-charts/stable/amazon-eks-pod-identity-webhook/templates/Service.yaml#L20

vivek-koppuru commented on helm-charts/stable/cloudwatch-agent/Chart.yaml:
```diff
@@ -0,0 +1,36 @@
+# Copyright 2020 Amazon.com Inc. or its affiliates. All Rights Reserved.
```
We are technically in the new year, might be worth changing to 2021? I don't know how these years work though

vivek-koppuru commented on helm-charts/stable/cloudwatch-agent/templates/cloudwatch-agent-daemonset.yaml:
```diff
@@ -0,0 +1,96 @@
+      containers:
+        - name: {{ include "cloudwatch-agent.fullname" . }}
+          image: {{ .Values.cloudwatchAgent.image }}
+          #ports:
+          #  - containerPort: 8125
+          #    hostPort: 8125
+          #    protocol: UDP
```
Any reason we leave it commented out then?

bnrjee replied:
From the original yaml. We can remove it.

vivek-koppuru commented on helm-charts/stable/cloudwatch-agent/templates/cloudwatch-agent-service-account.yaml:
```diff
@@ -0,0 +1,20 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{ if .Values.cloudwatchAgent.serviceAccount.create -}}
```
Does this service account need AWS iam role access? If not, then we can just create the service account through the helm chart instead of through the CDK.

bnrjee replied:
Yes, it requires, iam permissions.

vivek-koppuru commented on helm-charts/stable/cloudwatch-agent/values.yaml:
```diff
@@ -0,0 +1,27 @@
+clusterName: 'my-eks-cluster'
+
+cloudwatchAgent:
+  image: 'amazon/cloudwatch-agent:1.247346.0b249609'
```
Is this a valid image? If not, let's replace with a fake image like we do here: https://github.com/aws/eks-distro-build-tooling/blob/main/helm-charts/stable/amazon-eks-pod-identity-webhook/values.yaml#L22

bnrjee replied:
I think this is valid.
https://hub.docker.com/layers/amazon/cloudwatch-agent/1.247346.0b249609/images/sha256-ab6f27f16cfac6f2d307732fdf762f87baf57e2e4a530ec9e77a60855e8a99ea?context=explore

vivek-koppuru commented on helm-charts/stable/cloudwatch-agent/templates/cloudwatch-agent-daemonset.yaml:
```diff
@@ -0,0 +1,92 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "cloudwatch-agent.fullname" . }}
```
Add a section for labels here like the following example I commented above: https://github.com/aws/eks-distro-build-tooling/blob/main/helm-charts/stable/amazon-eks-pod-identity-webhook/templates/Service.yaml#L20

vivek-koppuru reviewed (approved):
/lgtm
/approve
